src/game/save_file.h

```diff
@@ -89,7 +91,7 @@ struct SaveBuffer {
     // Each of the four save files has two copies. If one is bad, the other is used as a backup.
     struct SaveFile files[NUM_SAVE_FILES][2];
     // The main menu data has two copies. If one is bad, the other is used as a backup.
-    struct MainMenuSaveData menuData[1];
+    struct MainMenuSaveData menuData[2];
```

Comment:
was this a bug fix?

Reply:
Yep, it broke menu saving, and I had two options:
Either accommodate menu saving to not backup data, or just force a minimum of EEP16K
I chose the latter.

Reply:
yup thats much better

Reply:
Doesn't that kind of suck for flashcart users? If you don't remember to set the save type on each boot, it won't save

Reply:
I think that only works for like new everdrives and not the ed64p or super64 though right?

Reply:
redefining to `EEP16K` might not make it in the rom header tho right?

Reply:
It does, this has been tested

Reply:
> It does, this has been tested

Just to be clear: You're sure that even shitty flashcarts like the ed64p will automatically detect that the savetype is EEP16K and use it without manual user intervention?

Reply:
It should, it's software level stuff.

Reply:
PJ64 was having issues with EEP16K the other day with Rovert which mimics issues I've had with it in the past as well. This needs an error screen by using the return value of osEepromProbe to check if the right eeprom size is set correctly.

Reply:
Also, 64Drive and older Everdrives don't respect the rom header settings. I doubt the ed64plus does either.

Reply:
Why are they terrible?

Reply:
> Why are they terrible?

It's what most people use so please make sure this works on ed64p and super64 and if it doesn't pls take an alternative route that allows using the vanilla save type :(

Reply:
oh or as wiseguy say, you could just show an error screen if the wrong save type is being used. There should never be a situation in where the game is simply silently failing to save, especially not out of the box with the default settings.

Reply:
I agree this should be a feature, although it should be treated independently of Puppyprint

Reply:
as far as errors go i think that we should display them as images so that way it's engine independent and can be easily changed if somebody wants to make them match their UI

Reply:
For anything that's absolutely mandatory you could boot into an error screen (idk if that applies here). If not mandatory, it should be a dismissable message.

Reply:
reonu created #130 

https://github.com/Reonu/HackerSM64/issues/130#issuecomment-992635059